apps/api/src/donations/donations.service.ts

```diff
@@ -370,6 +371,41 @@ export class DonationsService {
     return { donations, total }
   }
 
+  async getUserDonatedCampaigns(keycloakId: string) {
+    const donations = await this.prisma.donation.findMany({
+      where: { person: { keycloakId } },
```

Comment:
The approach is good, yet after looking in more detail, we can do it in more optimal way.
To return the campaigns to which the user has donated, we can use the same method that lists all campaigns and add a filter by user. 

See campaign.service.ts listCampaigns() function and either create a new function or just extend this one to accept an optional AuthorizedUser parameter and then add this to the where clause: vaults: {some: { donations: { some: {person: {keycloakId}}}}}, 

Also please add this into the other branch so that the related changes stay together.